### PR TITLE
Some performance tweaks for widgets

### PR DESCRIPTION
**Pull request by gebner**
The infoview has gotten noticably slower with large trace messages since last year.  It now seems roughly back to the state in December.  (No hard numbers, sorry.)

### CODE REVIEW

File: lua/lean/widgets.lua [hunk 1]
```diff
@@ -37,20 +34,33 @@ local BufRenderer = {
...
+
 ---Create a new Element.
+---@param args ElementNewArgs
 ---@return Element
-function Element:new(obj)
-  obj = obj or {}
-  local children = obj.children or {}
-  obj.children = nil
-  return setmetatable(
-    vim.tbl_extend("keep", obj, { events = {}, __children = children }),
```
Comment (gebner) [inline lua/lean/widgets.lua:47]: `tbl_extend` was 12% of the runtime btw.

Comment (Julian) [reply lua/lean/widgets.lua:47]: Eesh.

**Comment by Julian** [comment]
Thanks!

**Comment by Julian** [comment]
Is there a specific case you were performance optimizing with by the way? Or just something similar to what you've shared before that spits out 1000 trace messages? Just want to make sure I know what to check in the future.

**Comment by gebner** [comment]
Yes, it's just the one with the adjustable number of trace messages.  I should probably also look at a benchmark with collapsible traces.

```lean
def Good (n : Nat) := True

@[simp] theorem Good_zero : Good 0 := ⟨⟩
@[simp] theorem Good_succ_iff : Good (n+1) ↔ Good n := Iff.rfl

set_option trace.Meta.Tactic.simp true in
example : Good 5000 := by simp
```

**Pull request merged**